### PR TITLE
Fix lint jobs issues

### LINKED ISSUE #2948: Fix Lint job issues
The linter probably has got the subsequent update, we need to fix all the warnings found. See the Lint job comments to #2947.

### PR DESCRIPTION
**Pull request by ZhangTao1596**
Fix #2948 

1. ~~Add `go.work` to orginatize go modules~~
2. Ignore `revive` linter  `unused-parameter` check for all files ~~and `empty-block` check for all `doc_test.go`s~~.
3.  Fix ~~2 places~~ all `empty-block` issue
4. ~~Configuration to show all linter issues in ci~~

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -15,7 +15,7 @@ D_FILE ?= "$(shell if [ "$(OS)" = Windows_NT ]; then echo "Dockerfile.wsc"; else
 DC_FILE ?= ".docker/docker-compose.yml" # Single docker-compose for Ubuntu/WSC, should be kept in sync with ENV_IMAGE_TAG.
 ENV_IMAGE_TAG="env_neo_go_image"
 
-REPO ?= "$(shell go list -m)"
+REPO ?= "$(shell go list -m | head -n 1)"
```
Comment (roman-khimov) [inline Makefile:18]: That's something different.

File: .golangci.yml [hunk 2]
```diff
@@ -69,3 +69,13 @@ issues:
...
+  exclude-rules:
+    - path: doc_test.go
+      linters:
+        - revive
+      text: "empty-block"
+    - linters:
+        - revive
+      text: "unused-parameter"
+  max-issues-per-linter: 0
+  max-same-issues: 0
```
Comment (roman-khimov) [inline .golangci.yml:81]: These `max-*` look dangerous, I'd like to see everything there is from a linter to fix things in batches.

**Review by roman-khimov — COMMENTED** [review]
> Add go.work to orginatize go modules

Does it really help in our case? We've got multiple modules in the same repo.

> Ignore revive linter unused-parameter check for all files and empty-block check for all doc_test.gos.

Let's solve empty-block (I mean not including it into the ignore list, likely some code can be improved in examples too) and create an issue for unused-parameter.

> Fix 2 places empty-block issue

These are good.

> Still 3 places have empty block issue and all related channel clear. I think the code is concise no need to change for linter-pass purpose

`//nolint` can be used here.

File: Makefile (hunk 1 above)
Comment (ZhangTao1596) [reply Makefile:18]: With `go.work` this will list all go mods and the first one is neo-go mod.

File: .golangci.yml (hunk 2 above)
Comment (ZhangTao1596) [reply .golangci.yml]: Done.

**Comment by ZhangTao1596** [comment]
> Does it really help in our case? We've got multiple modules in the same repo.

I remove this because it requires go1.18 or later. Should we update go version in go.mods? We use 1.17 while 1.20.2 already released.
go.work can help to orgnize multi-mods project. We don't need `replace` in `go.mod` locally when change mutiple packages with dependencies. Besides my vscode linter doestn't work without it when edit nested module :).

**Comment by AnnaShaleva** [comment]
> Should we update go version in go.mods?

Not yet, we still have to support go 1.17 for some time. Although I think we'll deprecate it soon. The old go versions support is intentional and is mostly for our users.

**Review by AnnaShaleva — APPROVED** [review]

**Review by roman-khimov — APPROVED** [review]
> Should we update go version in go.mods?

We've got #2908, yes.

**Pull request merged**